Mathlib/Data/ZMod/Basic.lean

```diff
@@ -599,13 +618,41 @@ theorem val_add {n : ℕ} [NeZero n] (a b : ZMod n) : (a + b).val = (a.val + b.v
   · apply Fin.val_add
 #align zmod.val_add ZMod.val_add
 
+theorem val_add_of_lt {n : ℕ} [NeZero n] {a b : ZMod n} (h : a.val + b.val < n) :
+    (a + b).val = a.val + b.val := by rw [ZMod.val_add, Nat.mod_eq_of_lt h]
```

Comment:
Why does this need `[NeZero n]`? It's implied by `h`, right?

Reply:
Oh, yes, thanks, didn't catch that